assets/js/blocks/featured-product/block.js

```diff
@@ -28,178 +26,45 @@ import {
 	withSpokenMessages,
 } from '@wordpress/components';
 import classnames from 'classnames';
-import { Component, Fragment } from '@wordpress/element';
+import { Fragment } from '@wordpress/element';
 import { compose } from '@wordpress/compose';
-import { debounce, isObject, isEmpty } from 'lodash';
+import { isEmpty } from 'lodash';
 import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import ProductControl from '../../components/product-control';
 import ApiErrorPlaceholder from '../../components/api-error-placeholder';
+import {
+	dimRatioToClass,
+	getBackgroundImageStyles,
+} from './utils';
 import {
 	getImageSrcFromProduct,
 	getImageIdFromProduct,
 } from '../../utils/products';
+import withProduct from '../../utils/with-product';
```

Comment:
I wonder if we should put hocs in a top-level (living at the same level as `components`) folder?  In the pull I'm working on I have `js/hocs/with-searched-products.js`.  I don't think we should initially expose hocs publicly yet, but it does provide the option down the road for exposing them on their own package (and makes it easier to locate them too).

Reply:
Absolutely. Done in 72abc254e72d928f21035f87bd0852da5174aef3.

We will need to move `with-component-id.js` in a follow-up too.

Reply:
> In a follow-up, we will need to move with-component-id.js too.

Done in #797.